src/hotspot/share/gc/shenandoah/mode/shenandoahMode.hpp

```diff
@@ -32,16 +32,16 @@ class ShenandoahHeuristics;
 #define SHENANDOAH_CHECK_FLAG_SET(name)                                     \
   do {                                                                      \
     if (!(name)) {                                                          \
-      err_msg message("GC mode needs -XX:+" #name " to work correctly");    \
-      vm_exit_during_initialization("Error", message);                      \
+      const char *msg = "GC mode needs -XX:+" #name " to work correctly";   \
```

Comment:
Please decide which way the `*` leans in this change. I prefer `const char* msg`, like in the change below.

Reply:
In fact, maybe just inline this literal down in `vm_exit_during_initialization` invocation.

Reply:
Of course it should align to the left, like in the rest of hotspot. Thanks for noticing!

As for inlining of the message, there are pros and cons. The arguments should be aligned, so it would become
```
  do {                                                                      \
    if (!(name)) {                                                          \
      vm_exit_during_initialization("Error", msg);                          \
                                    "GC mode needs -XX:+" #name " to work correctly"; \
    }                                                                       \
  } while (0)

```
(not pretty at all, you see).

After that, there is an option to split the string into multiple lines in attempt to shrink the length, but then the line would become ungreppable (much bigger evil).

Among options, I decided to respect intention and style of original author, who introduced a variable for err_msg, which is unusual and should have some valid rationale behind, like the one above.

Reply:
I _am_ the author of those `SHENANDOAH_CHECK_FLAG_SET` blocks ;)

This should be fine:

```
do {                                                     \
  if (!(name)) {                                         \
    vm_exit_during_initialization("Error",               \
      "GC mode needs -XX:+" #name " to work correctly"); \
  }                                                      \
 } while (0)
```

Reply:
OK, if you're OK with that. But let's introduce some consistency with e.g. https://github.com/openjdk/jdk/blob/a804c6a6/src/hotspot/share/gc/shenandoah/shenandoahHeap.cpp#L420 and use 8 spaces for the argument indentation

```
  do {                                                                      \
    if (!(name)) {                                                          \
      vm_exit_during_initialization("Error",                                \
              "GC mode needs -XX:+" #name " to work correctly");            \
    }                                                                       \
  } while (0)
```

Reply:
Fine with me!